### PR TITLE
Introducing Dependabot grouped updates

### PR DESCRIPTION
**Pull request by lelia**
## Description

Dependabot has [rolled out a public beta](https://github.blog/changelog/2023-06-30-grouped-version-updates-for-dependabot-public-beta/) for custom grouping rules based on pattern-matching package names. Certain patterns [can also be excluded](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#groups) (although I wasn't sure if any exclusions would apply to this project, so left alone for now).

### Adds

* Grouped updates for `pip` (python) and `github-actions` packages
  * ⚠️ Currently in public beta / subject to change

### Removes

* (_Seemingly_) unused `docker` rules from Dependabot config
  * 🐳 Let me know if this should stay put!

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

### CODE REVIEW

File: .github/dependabot.yml
Comment (agentlecisco) [inline .github/dependabot.yml:9]: Yes, I think it's fine to remove Docker dependencies.

**Review by agentlecisco — APPROVED** [review]
Looks good, let's give it a try.

**Comment by lelia** [comment]
> > Looks good, let's give it a try.
> 
> Thanks @annegentle! For some reason, it's still showing merging as blocked even though you approved and appear to be a member of `inclusive-language-maintainers`. Does it require a second approval?

Actually, I just realized what happened here - looks like you approved the PR from your @agentlecisco which is _not_ a member of the maintainer group 🙂 Mind re-approving from @annegentle ?

**Review by agentlecisco — APPROVED** [review]
Ha, I keep doing that, approving with the `agentlecisco` account.

**Review by annegentle — APPROVED** [review]
👍

**Comment by annegentle** [comment]
@lelia  when you get a chance, please add agentlecisco and remove annegentle.

**Pull request merged**
